src/python-common/ceph/deployment/drive_group.py

```diff
@@ -255,10 +255,10 @@ def validate(self):
         if self.objectstore not in ('filestore', 'bluestore'):
             raise DriveGroupValidationError("objectstore not in ('filestore', 'bluestore')")
 
-        if self.block_wal_size is not None and type(self.block_wal_size) != int:
-            raise DriveGroupValidationError('block_wal_size must be of type int')
-        if self.block_db_size is not None and type(self.block_db_size) != int:
-            raise DriveGroupValidationError('block_db_size must be of type int')
+        for name in 'block_wal_size block_db_size journal_size'.split():
+            val = getattr(self, name)
+            if val is not None and not isinstance(val, (int, float)):
+                raise DriveGroupValidationError(f'{name} must be of type int or of type float')
```

Comment:
Could you do `f'{name} must be of type int or of type float but is currently {val}'` so it's clear what it actually is?